File: April 2020/15 - Product of Array Except Self/Solution.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = [1] * len(nums)
        
        L = 1
        R = 1
        
        for i in range(len(nums)):
            ans[i] *= L;
            ans[-1-i] *= R;
            L *= nums[i]
            R *= nums[-1-i]
        
        return ans

File: April 2020/15 - Product of Array Except Self/Solution.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans(nums.size(), 1);
        int L, R;
        L = R = 1;
        for(int i = 0, j = nums.size()-1; i < nums.size() && j >= 0; ++i, --j) {
            ans[i] *= L;
            ans[j] *= R;
            L *= nums[i];
            R *= nums[j];
        }
        return ans;
    }
};